src/platform/platform_posix.c

```diff
@@ -460,8 +460,13 @@ CxPlatProcCurrentNumber(
             "=c" (cpuinfo[2]),
             "=d" (cpuinfo[3])
             : "a"(1));
-    CXPLAT_FRE_ASSERT((cpuinfo[3] & (1 << 9)) != 0);
-    return (uint32_t)cpuinfo[1] >> 24;
+    //
+    // Cannot be an assert, as M1 running under Rosetta always has this flag 0.
+    //
```

Comment:
Instead of comment why this can't be an assert (which makes no sense to anyone who hasn't watched the history of this code evolve) please update the comment to say why/what is happening below.